File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/DataType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Linq;

namespace WPFComponents
{
  public class DataType
  {
    public string Name { get; set; }

    public IEnumerable<DataType> GetDataTypes(string languageName)
    {
      XElement xe;
      IEnumerable<DataType> items = null;
       
      if (languageName == "C#")
        xe = XElement.Load(WPFCommon.GetCurrentDirectory() + @"\Xml\DataTypesCS.xml");
      else if (languageName == "VB.NET")
        xe = XElement.Load(WPFCommon.GetCurrentDirectory() + @"\Xml\DataTypesVB.xml");
      else
        xe = null;

      // Always check for null because of design time problem
      if (xe != null)
      {
        items = (from item in xe.Descendants("DataType")
                     select new DataType
                     {
                       Name = item.Element("Name").Value
                     });
      }

      return items;
    }
  }
}

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/VB/WPFListBox-VB/WPFComponents/DataType.vb
Public Class DataType
  Private mName As String

  Public Property Name() As String
    Get
      Return (mName)
    End Get
    Set(ByVal value As String)
      mName = value
    End Set
  End Property

  Public Function GetDataTypes(ByVal languageName As String) As IEnumerable(Of DataType)
    Dim xe As XElement
    Dim items As IEnumerable(Of DataType) = Nothing

    If languageName = "C#" Then
      xe = XElement.Load(WPFCommon.GetCurrentDirectory() & "\Xml\DataTypesCS.xml")
    ElseIf languageName = "VB.NET" Then
      xe = XElement.Load(WPFCommon.GetCurrentDirectory() & "\Xml\DataTypesVB.xml")
    Else
      xe = Nothing
    End If

    ' Always check for null because of design time problem
    If xe IsNot Nothing Then
      items = (From item In xe.Descendants("DataType") _
           Select New DataType())
    End If

    Return (items)
  End Function
End Class